packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1034,6 +1082,28 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
     }
   }
 
+  private getHighlightedIndexOnCharKeyDown = (keyString: string): number => {
```

Comment:
I am lost with semantics of this method, it's called `getHighlightedIndexOnCharKeyDown()`, but it contains `setStartingString`

Should it be `setHighlightedIndexOnCharKeyDown()` and contain `this.setState({ highlightedIndex })`?

Reply:
good point. I will refactor it as a setter. Let's see if I also need to refactor the setStartingString one with debounce or not.